### PR TITLE
Update product placeholders with products from design

### PR DESCRIPTION
**Pull request by albarin**
## What

This PR updates the dummy products to use real product info and images.

It also reduces the size of one of the images, it was huge in px and close to 1 Mb.
Fixes https://github.com/woocommerce/woocommerce-blocks/pull/11997#pullrequestreview-1757905545

## Why

To have real product placeholders instead of dummy empty ones.

## Testing Instructions

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. On a JN or WooExpress installation, replace the WooCommerce Blocks plugin with this one: https://wcblocks.wpcomstaging.com/wp-content/uploads/woocommerce-gutenberg-products-block-12007.zip
2. Using the plugin editor comment out [this line](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/src/StoreApi/Routes/V1/AI/Product.php#L83) to avoid replacing the dummy products with AI-generated ones.
3. Go through the CYS process.
4. Make sure all the dummy products were created with their corresponding title, description, image and price.
5. Uncomment the line commented out in step 2, reset the CYS and delete all products on `wp-admin/tools.php?page=woocommerce-admin-test-helper`.
6. Go through the CYS process and make sure the products are AI-generated ones and no dummy products are shown.

* [x] Do not include in the Testing Notes 
* [x] Should be tested by the development team exclusively 

## Screenshots or screencast

| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility

Required:

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [ ] N/A

## Checklist

Required:

* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [ ] This PR is assigned to a milestone.

Conditional:

* [ ] This PR has a UI change and has been cross-browser tested at different viewport sizes on both the frontend and in the editor.
* [x] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

### CODE REVIEW

File: src/Patterns/ProductUpdater.php [hunk 1]
```diff
@@ -8,6 +8,44 @@
  * Pattern Images class.
  */
 class ProductUpdater {
+	const DUMMY_PRODUCTS = [
+		[
+			'title'       => 'Vintage Typewriter',
+			'image'       => 'images/pattern-placeholders/watch-hand-brand-jewellery-strap-platinum.jpg',
```
Comment (nefeline) [inline src/Patterns/ProductUpdater.php]: It looks like the image here is incorrect: can you take a look & confirm, please? For Vintage Typewriter, I would expect the following image:

Screenshot 2023-12-01 at 16 19 50

**Review by nefeline — COMMENTED** [review]
Great work!

Confirmed all dummy products are created with the correct content as expected:

Screenshot 2023-12-01 at 16 17 01

Screenshot 2023-12-01 at 16 24 01

Also confirmed these products are updated by AI when the feature is enabled.

I just left a comment regarding the image for the Vintage Typewriter product.

File: src/Patterns/ProductUpdater.php (hunk 1 above)
Comment (albarin) [reply src/Patterns/ProductUpdater.php]: Of course! Replaced the image here https://github.com/woocommerce/woocommerce-blocks/pull/12007/commits/70a62e2b7811f0240815815fb275e1cfb4aa5892

**Review by nefeline — APPROVED** [review]
Thanks for addressing the feedback. Works like a charm!

**Comment by nefeline** [comment]
Registering here for our future reference that the content and images used to update the products on this PR were obtained from the following design for the Product Gallery pattern (which is powered by the Product Collection block): FlPl7fHaGv9mpu8Nfl0joH-fi-158_32452

Screenshot 2023-12-01 at 16 18 14

**Pull request merged**